src/libinjector/linux/methods/linux_shellcode.c

```diff
@@ -0,0 +1,164 @@
+#include "linux_shellcode.h"
+#include "linux_debug.h"
+#include "linux_syscalls.h"
+
+bool setup_mmap_trap(drakvuf_t drakvuf, drakvuf_trap_info_t* info);
+bool write_shellcode_to_mmap_location(drakvuf_t drakvuf, drakvuf_trap_info_t* info);
+
+event_response_t handle_shellcode(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+
+    injector_t injector = (injector_t)info->trap->data;
+
+    event_response_t event;
+
+    switch (injector->step)
+    {
+        case STEP1: // Finds vdso and sets up mmap
+        {
+            memcpy(&injector->saved_regs, info->regs, sizeof(x86_registers_t));
+
+            addr_t vdso = find_vdso(drakvuf, info);
+            addr_t syscall_addr = find_syscall(drakvuf, info, vdso);
+
+            setup_post_syscall_trap(drakvuf, info, syscall_addr);
+            // don't remove the initial trap
+            // it is used for cleanup after restoring registers
+
+            if (!setup_mmap_syscall(injector, info->regs, 4096))
+                PRINT_DEBUG("Failed to setup mmap syscall");
+
+            info->regs->rip = syscall_addr;
+            info->regs->rax = injector->syscall_no;
+
+            event = VMI_EVENT_RESPONSE_SET_REGISTERS;
+
+            break;
+        }
+        case STEP2: // setup shellcode
+        {
+            PRINT_DEBUG("mmap location is: %lx\n", info->regs->rax);
```

Comment:
I don't think this is `mmap` location, but rather an address of the memory allocated using `mmap`.

Reply:
then should i change the statement to `memory address allocated using mmap: %lx\n`?